dlt/common/plugins/plugins.py

```diff
@@ -0,0 +1,55 @@
+from typing import Type
+from dlt.common.typing import TDataItem
+from dlt.common.pipeline import SupportsPipeline
+from dlt.common.configuration.specs import ContainerInjectableContext
+from dlt.common.schema.typing import TSchemaContract
+from dlt.common.configuration.specs import configspec
+
+
+class Plugin:
```

Comment:
we could consider having multiple plugin base classes which each implement a set of related hooks. This way we can group them in the pluginsprovider and not have to iterate all of the for filtering rows, only the relevant ones. tbd.